runtime/ms_rest_azure/lib/ms_rest_azure/azure_service_client.rb

```diff
@@ -41,7 +41,7 @@ def get_long_running_operation_result(azure_response, custom_deserialization_blo
       if !AsyncOperationStatus.is_terminal_status(polling_state.status)
         task = Concurrent::TimerTask.new do
           begin
-            if !polling_state.azure_async_operation_header_link.nil?
+            if !polling_state.azure_async_operation_header_link.nil? && custom_deserialization_block.nil?
```

Comment:
@sarangan12 can you describe a little more how this additional condition helps with not returning nil values?